### DISCUSSION TITLE
allure-js-commons V1 us unsupported but used by ConceptJS

### DISCUSSION THREAD

**Discussion by dzienisz** [allure-framework/allure2#1763] [Bug Reports] [answered]
- Is is possible to update `allure-js-commons` dependencies? 
- When `allure-js-commons` v2 will be available?
- What is the future of `allure-js-commons`?
- [ConceptJS](https://github.com/codeceptjs/CodeceptJS) use it and don't know is there any library that they can switch to?

**Comment by baev** [comment]
We suggest everyone to switch to new [Allure JS integrations](https://github.com/allure-framework/allure-js). 

We release it in beta channel as we want to be able to introduce breaking changes to commons package in any moment. The main idea is that we are trying to align API of commons module with API provided by other languages and make it easier to integrate with frameworks that support parallel modes. And we don't want to release new major version per such change as we want to have the same release version for all the Allure packages (like Allure Java has the same version as Allure Report).

So the global strategy is keep adding official integrations with other frameworks and collect all the pain points (what is missed, what is hard to implement or support) and immediately introduce desired API changes to commons.

It shouldn't be a problem for end-users though, but could cause some issues for people who maintain own Allure Integration. 

Is there any issues with ConceptJS you want to address? 

BTW I do believe we may land stable release version for allure-js-commons in 2022

**Comment by baev** [comment, marked as answer]
Update from 2024:

1. CodeceptJS is now supported officially by the Allure team. Docs https://allurereport.org/docs/codeceptjs/ , Npm package https://www.npmjs.com/package/allure-codeceptjs
3. allure-js-commons is out of beta
